machine/machine_core/ssh_connection.py

```diff
@@ -56,6 +56,13 @@ def message(self, *args):
         if not self.verbose:
             return
         print(" ".join(args))
+        # need to flush stdout immediately to avoid delaying the message with stderr output when writing to log files
+        while True:
```

Comment:
Is it possible that this would get stuck indefinitely? If there is slight chance then maybe do this waiting N times and then give up?

Reply:
That's the code that we've been using in run-tests for a long time (I copied it from there)

Doesn't make much of a difference, though: If the log sink is broken (stdout indefinitely blocked), then the test log will be truncated and corrupted either way. There is no point in running the tests further. The usual result is that the task container aborts the run because the sink's ssh exits with 255; but even if not, we still have the global timeout to make sure that nothing gets stuck indefinitely.

We could "optimize" the timeout by trying to flush for up to, say, 30 seconds, and then raising the exception. I can send a follow-up for that (also for run-tests) if you want.